core/src/main/java/com/cloud/agent/transport/ArrayTypeAdaptor.java

```diff
@@ -75,13 +75,17 @@ public T[] deserialize(JsonElement json, Type typeOfT, JsonDeserializationContex
             try {
                 clazz = Class.forName(name);
             } catch (ClassNotFoundException e) {
-                throw new CloudRuntimeException("can't find " + name);
+                throw new JsonParseException("can't find " + name);
             }
             T cmd = (T)_gson.fromJson(entry.getValue(), clazz);
             cmds.add(cmd);
         }
-        Class<?> type = ((Class<?>)typeOfT).getComponentType();
-        T[] ts = (T[])Array.newInstance(type, cmds.size());
-        return cmds.toArray(ts);
+        try {
+            Class<?> type =  Class.forName(typeOfT.getTypeName().replace("[]", ""));
+            T[] ts = (T[])Array.newInstance(type, cmds.size());
+            return cmds.toArray(ts);
+        } catch (ClassNotFoundException e) {
+            throw new CloudRuntimeException("can't find " + typeOfT.getTypeName());
```

Comment:
Should this also be `JsonParseException`?

Reply:
I remember there was a reason, but do not remember clearly. let me have try